src/index.tsx

```diff
@@ -25,12 +25,16 @@ import { getApi, setApi } from '@/api/Api'
 import MapActionReceiver from '@/stores/MapActionReceiver'
 import { createMap, getMap, setMap } from '@/map/map'
 
-let locale = new URL(window.location.href).searchParams.get('locale')
+const url = new URL(window.location.href)
+const locale = url.searchParams.get('locale')
 setTranslation(locale || navigator.language)
 
-// set up state management
-setApi(config.api, getApiKey())
-const queryStore = new QueryStore(getApi())
+// use graphhopper api key from url or try using one from the config
+const apiKey = url.searchParams.has('key') ? url.searchParams.get('key') : config.keys.graphhopper
+setApi(config.api, apiKey)
+
+const initialCustomModelStr = url.searchParams.get('custom_model')
```

Comment:
We take the custom model from the url, but then we set the url from our state somewhere, so the custom_model parameter disappears. Should we add it to the url parameters? And even keep it updated the whole time? Not sure...

Reply:
Looks like this is easy (?): c96eda0d9ec89f8bc72f9e352f89bcc155d3b5af
Can you check if this is what you want? And maybe we need some url encoding or something?